apps/ejabberd/src/mod_roster.erl

```diff
@@ -781,28 +775,17 @@ send_unsubscription_to_rosteritems(LUser, LServer) ->
                   RosterItems).
 
 %% @spec (From::jid(), Item::roster()) -> ok
-send_unsubscribing_presence(From, Item) ->
-    IsTo = case Item#roster.subscription of
-               both -> true;
-               to -> true;
-               _ -> false
-           end,
-    IsFrom = case Item#roster.subscription of
-                 both -> true;
-                 from -> true;
-                 _ -> false
-             end,
-    if IsTo ->
-           send_presence_type(jid:to_bare(From),
-                              jid:make(Item#roster.jid),
-                              <<"unsubscribe">>);
-       true -> ok
+send_unsubscribing_presence(From, #roster{ subscription = Subscription } = Item) ->
```

Comment:
I partially agree with @arcusfelis here. Maybe leaving `IsTo` and `IsFrom` vars will make it easier to understand. Of course new way of deciding on the value is far better.